### PR TITLE
Refactor rpc utils test

### PR DESCRIPTION
**Pull request by gzhao9**
## What is the purpose of the change

Similarly, to #13153, I noticed that in `RpcUtilsTest`, there is also a duplicate creation of the mock.

The `Invoker` was duplicated to create the mock object 10 times, in 10 test cases.
I used a similar approach refactoring to increase code readability.

## Brief changelog

I refactored the code to introduce two methods: `CreateMockInvoker(URL url)` and `CreateMockInvoker()`. I've used these methods in 10 test cases to reduce the repetitive creation of the `Invoker` mock.

### Additional questions.
I also noticed that this pattern appears in different test files for different packages, do you prefer to use shared utility classes or define these methods in each file?

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/support/RpcUtilsTest.java [hunk 1]
```diff
@@ -50,6 +50,14 @@ class RpcUtilsTest {
      * regular scenario: async invocation in URL
      * verify: 1. whether invocationId is set correctly, 2. idempotent or not
      */
+    Invoker CreateMockInvoker(URL url){
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/support/RpcUtilsTest.java:53]:
```suggestion
    Invoker createMockInvoker(URL url){
```

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/support/RpcUtilsTest.java [hunk 2]
```diff
@@ -50,6 +50,14 @@ class RpcUtilsTest {
      * regular scenario: async invocation in URL
      * verify: 1. whether invocationId is set correctly, 2. idempotent or not
      */
+    Invoker createMockInvoker(URL url){
+        Invoker invoker = CreateMockInvoker();
+        given(invoker.getUrl()).willReturn(url);
+        return  invoker;
+    }
+    Invoker CreateMockInvoker(){
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/support/RpcUtilsTest.java:58]:
```suggestion
    Invoker createMockInvoker(){
```

**Review by AlbumenJ — COMMENTED** [review]
Also fix in your code

File: dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/support/RpcUtilsTest.java (hunk 2 above)
Comment (gzhao9) [reply dubbo-rpc/dubbo-rpc-api/src/test/java/org/apache/dubbo/rpc/support/RpcUtilsTest.java]: I have replaced all `CreateMockInvoker` with `createMockInvoker`

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**